codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -399,7 +365,20 @@ public Http2Stream close() {
             }
 
             state = CLOSED;
-            deactivateInternal(this);
+            if (activeStreams.remove(this)) {
+                try {
+                    // Update the number of active streams initiated by the endpoint.
+                    createdBy().numActiveStreams--;
+
+                    // Notify the listeners.
+                    for (Listener listener : listeners) {
+                        listener.streamClosed(this);
```

Comment:
But in this case since the `try { ...}` was added to the outer scope we may just want to remove it from this PR and address all instances of this issue in another PR...or just address them all in this PR?